rasa/shared/core/domain.py

```diff
@@ -651,11 +667,20 @@ def user_actions_and_forms(self):
         return self.user_actions + self.form_names
 
     @rasa.shared.utils.common.lazy_property
-    def num_actions(self):
-        """Returns the number of available actions."""
+    def action_names(self) -> List[Text]:
+        """Returns action names or texts."""
+        rasa.shared.utils.io.raise_warning(
```

Comment:
Do we not want `raise_deprecation_warning`?

Reply:
@wochinge could you please take this over

Reply:
No, because this is something which only affects developers. `raise_deprecation_warning` will use a `FutureWarning` which will be printed. However, it shouldn't be printed as our code will only use the new things. Users using the Python API should receive a deprecation message though. Will add a comment